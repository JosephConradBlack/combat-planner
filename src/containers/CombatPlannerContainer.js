import { connect } from 'react-redux';

import CombatPlannerPage from '../components/CombatPlannerPage';

import { editFighter, clearFighters } from '../actions/fighterActions';
import { getFighters } from '../selectors/fighterSelectors';

const mapStateToProps = state => {
  return {
    items: getFighters(state)
  };
};

const mapDispatchToProps = {
  onFighterUpdate: editFighter,
  onClearFighters: clearFighters
};

const CombatPlannerContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(CombatPlannerPage);

export default CombatPlannerContainer;
