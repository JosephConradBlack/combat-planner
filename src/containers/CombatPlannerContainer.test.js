import React from 'react';
import { create } from 'react-test-renderer';

import CombatPlannerContainer from './CombatPlannerContainer';

describe('CombatPlannerContainer', () => {
  it('renders ', () => {
    const component = create(<CombatPlannerContainer />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
