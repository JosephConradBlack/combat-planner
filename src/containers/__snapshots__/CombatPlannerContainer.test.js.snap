// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CombatPlannerContainer renders  1`] = `
<div>
  <div
    className="ui inverted vertical center aligned segment sc-bdVaJa lasqMc"
  >
    <div
      className="ui large inverted pointing secondary menu"
    >
      <div
        className="ui container"
      >
        <div
          className="item"
          onClick={[Function]}
        >
          Combat Planner
        </div>
      </div>
    </div>
    <div
      className="ui container"
    >
      <h1
        className="ui inverted header sc-bwzfXH cKpNQC"
      >
        Combat Planner
      </h1>
    </div>
  </div>
  <div
    className="ui basic segment sc-htoDjs cLBMIL"
  >
    <div
      className="ui container sc-dnqmqq gHNiFF"
    >
      <div
        className="ui segments"
      >
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  1
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Player 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  2
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Player 2
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  3
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Orc 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  4
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Goblin 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  5
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Goblin 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  6
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Goblin 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  7
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Goblin 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  8
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Goblin 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  9
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Goblin 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  10
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Goblin 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui segment"
        >
          <div
            className="item"
          >
            <div
              className="content"
            >
              <div
                className="extra"
              >
                <div
                  className="ui left floated header"
                >
                  11
                </div>
              </div>
              <div
                className="header sc-gzVnrw JWpAx"
              >
                Goblin 1
              </div>
              <div
                className="extra sc-bZQynM iEOshs"
                position={1}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="red heart outline big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX kOPaAK"
                      color="red"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="extra sc-bZQynM fmGHdG"
                position={0}
              >
                <div>
                  <div
                    className="extra sc-htpNat fSoQxP"
                  >
                    <i
                      aria-hidden="true"
                      className="grey shield big icon sc-bxivhb iQXIBe"
                    />
                    <div
                      className="sc-ifAKCX ipPYkA"
                      color="grey"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
